Make the s2i-custom-notebook image a builder image

4n4nd opened the pull request:
Add Dockerfile to build Image
Add Makefile
Remove script copying from assemble and move it to Dockerfile
Update Python version to 3.6 in the demo Pipfile

Commit:
Update pipfile python version

4n4nd commented on Dockerfile:
```diff
@@ -0,0 +1,26 @@
+# s2i-custom-notebook
+FROM quay.io/thoth-station/s2i-minimal-notebook:master
```
@harshad16 is this the correct base image to use?

harshad16 replied:
yes it the correct base image :+1:

harshad16 commented on Makefile:
```diff
@@ -0,0 +1,11 @@
+
```
we would use thoth-ci to build this image, so it would directly build from Dockerfile.
Makefile is not needed if it is not doing anything more the build.WDYT?

4n4nd replied:
yeah we don't need it, it's just convenient when building the image locally

harshad16 commented on Dockerfile:
```diff
@@ -0,0 +1,26 @@
+RUN mv /opt/app-root/builder/run /opt/app-root/builder/run.original
+
+# Copy the builder files into /opt/app-root
+COPY ./.s2i/bin/* /opt/app-root/builder/
```
as .s2i is not being used in a conventional way, maybe renaming the dir to s2i-custom-notebook wouldn't raise any confusion. WDYT?

4n4nd replied:
in that case we can rename the directory to `s2i` instead, since that's what was generated when I used `s2i create` for this project.

harshad16 replied:
yeah any name which doesn't mislead developers thinking this is a conventional source2image build. :+1:

4n4nd replied:
makes sense :100:

Commit:
Add Dockerfile to build Image
Remove script copying from assemble and move it to Dockerfile 
the built image can be used as a builder image now
rename `.s2i` to `s2i`
Update README.md

harshad16 commented:
let us merge this by end of the day. 
As this is related-to jupyter notebook build pipeline as well.

harshad16 reviewed (approved):
LGTM, thanks :)
